packages/block-editor/src/hooks/typography.js

```diff
@@ -76,159 +42,93 @@ export const TYPOGRAPHY_SUPPORT_KEYS = [
 	LETTER_SPACING_SUPPORT_KEY,
 ];
 
-export function TypographyPanel( props ) {
-	const { clientId } = props;
-	const isFontFamilyDisabled = useIsFontFamilyDisabled( props );
-	const isFontSizeDisabled = useIsFontSizeDisabled( props );
-	const isFontAppearanceDisabled = useIsFontAppearanceDisabled( props );
-	const isLineHeightDisabled = useIsLineHeightDisabled( props );
-	const isTextDecorationDisabled = useIsTextDecorationDisabled( props );
-	const isTextTransformDisabled = useIsTextTransformDisabled( props );
-	const isLetterSpacingDisabled = useIsLetterSpacingDisabled( props );
+function TypographyInspectorControl( { children } ) {
+	return (
+		<InspectorControls __experimentalGroup="typography">
+			{ children }
+		</InspectorControls>
+	);
+}
 
-	const hasFontStyles = ! useIsFontStyleDisabled( props );
-	const hasFontWeights = ! useIsFontWeightDisabled( props );
+export function TypographyPanel( {
+	clientId,
+	name,
+	attributes,
+	setAttributes,
+} ) {
+	const settings = {
+		typography: {
+			fontFamilies: {
+				custom: useSetting( 'typography.fontFamilies' ),
+			},
+			fontSizes: {
+				custom: useSetting( 'typography.fontSizes' ),
+			},
+			fontStyle: useSetting( 'typography.fontStyle' ),
+			fontWeight: useSetting( 'typography.fontWeight' ),
+			lineHeight: useSetting( 'typography.lineHeight' ),
+			textDecoration: useSetting( 'typography.textDecoration' ),
+			textTransform: useSetting( 'typography.textTransform' ),
+			letterSpacing: useSetting( 'typography.letterSpacing' ),
+		},
+	};
 
-	const isDisabled = useIsTypographyDisabled( props );
-	const isSupported = hasTypographySupport( props.name );
+	const isSupported = hasTypographySupport( name );
+	const isEnabled = useHasTypographyPanel( name, null, settings );
 
-	if ( isDisabled || ! isSupported ) return null;
+	const value = useMemo( () => {
+		return {
+			...attributes.style,
+			typography: {
+				...attributes.style?.typography,
+				fontFamily: attributes.fontFamily
+					? 'var:preset|font-family|' + attributes.fontFamily
+					: undefined,
+				fontSize: attributes.fontSize
+					? 'var:preset|font-size|' + attributes.fontSize
+					: attributes.style?.typography?.fontSize,
+			},
+		};
+	}, [ attributes.style, attributes.fontSize, attributes.fontFamily ] );
```

Comment:
This is actually an interesting code. It highlights clearly the inconsistencies we have between the way we persist styles between "global styles" and "block attributes". 

My ideal would be to unify both long term but this involves removing block attributes and deprecations which is not super fun. But right now, I believe we could have two functions: `attributesToGlobalStyle` and `globalStyleToAttributes` and we could extract them from these "hooks" file into a dedicated tested code. 

I think it's fine to start like that though until we add all the other panels.

Reply:
Unifying and removing classes can simplify the code, but it can also limit flexibility for users. For example, currently, users can use a class to automatically apply a different hover color to a button that has a specific color class. If the classes are removed, this would no longer be possible.

Reply:
I'm not proposing to remove the classes here, I'm proposing to remove the attributes but store the value in `style` attribute as `car|preset|color|something`

Reply:
It's fewer differences than I thought there'd be 😅

Agree about storing `var|preset` in `style`. Would clean this up beautifully.